### PR TITLE
Mimic odo cli organization, add config command

### PR DESCRIPTION
**Pull request by johnmcollier**
Signed-off-by: John Collier <John.J.Collier@ibm.com>

## What is the purpose of this change? What does it change?
* Re-organizes the `kdo-poc` branch so that it mimics the ODO codebase
* Copies over the `odo config` command for use in `kdo`
* Copies over some applicable unittests from odo

### CODE REVIEW

File: pkg/kdo/cli/config/set.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+// Complete completes SetOptions after they've been created
+func (o *SetOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+
+	if o.envArray == nil {
+		o.paramName = args[0]
+		o.paramValue = args[1]
+	}
+	//o.context = genericclioptions.NewContext(cmd)
```
Comment (rajivnathan) [inline pkg/kdo/cli/config/set.go]: Should this comment be removed?

File: pkg/kdo/cli/config/unset.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+	"github.com/redhat-developer/odo-fork/pkg/kdo/genericclioptions"
+
+	"github.com/redhat-developer/odo-fork/pkg/kdo/cli/ui"
+	"github.com/redhat-developer/odo-fork/pkg/log"
+
+	"github.com/pkg/errors"
+	"github.com/redhat-developer/odo-fork/pkg/config"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
```
Comment (rajivnathan) [inline pkg/kdo/cli/config/unset.go]: Same comment. The original version from odo used the "k8s.io/kubernetes/pkg/kubectl/cmd/templates" import. Is there a reason we're switching to k8s.io/kubectl/pkg/util/templates?

File: pkg/kdo/cli/config/unset.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+func NewUnsetOptions() *UnsetOptions {
+	return &UnsetOptions{}
+}
+
+// Complete completes UnsetOptions after they've been created
+func (o *UnsetOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	if o.envArray == nil {
+		o.paramName = args[0]
+	}
+	//o.context = genericclioptions.NewContext(cmd)
```
Comment (rajivnathan) [inline pkg/kdo/cli/config/unset.go]: Should this comment be removed?

File: pkg/kdo/cli/config/view.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/redhat-developer/odo-fork/pkg/config"
+	"github.com/redhat-developer/odo-fork/pkg/kdo/genericclioptions"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
```
Comment (rajivnathan) [inline pkg/kdo/cli/config/view.go]: Same comment. The original version from odo used the "k8s.io/kubernetes/pkg/kubectl/cmd/templates" import. Is there a reason we're switching to k8s.io/kubectl/pkg/util/templates?

File: pkg/kdo/cli/config/config.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/redhat-developer/odo-fork/pkg/config"
+	"github.com/redhat-developer/odo-fork/pkg/kdo/util"
+
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/kdo/cli/config/config.go]: Newer client-go versions moved that file to that location

File: pkg/kdo/cli/config/set.go (hunk 1 above)
Comment (johnmcollier) [reply pkg/kdo/cli/config/set.go]: The comment needs to stay until we implement the `kclient` package and replace occlient

File: pkg/kdo/cli/config/unset.go (hunk 3 above)
Comment (johnmcollier) [reply pkg/kdo/cli/config/unset.go]: The comment needs to stay until we implement the kclient package and replace occlient

File: pkg/kdo/cli/config/set.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+	"github.com/redhat-developer/odo-fork/pkg/kdo/cli/ui"
+
+	"github.com/pkg/errors"
+	"github.com/redhat-developer/odo-fork/pkg/config"
+	"github.com/redhat-developer/odo-fork/pkg/kdo/genericclioptions"
+	"github.com/redhat-developer/odo-fork/pkg/log"
+
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
```
[the remark this answers is not in the document]
Comment (rajivnathan) [reply pkg/kdo/cli/config/set.go]: See previous comment for resolution.

File: pkg/kdo/cli/config/view.go (hunk 4 above)
Comment (rajivnathan) [reply pkg/kdo/cli/config/view.go]: See previous comment for resolution

File: pkg/kdo/cli/config/unset.go (hunk 2 above)
Comment (rajivnathan) [reply pkg/kdo/cli/config/unset.go]: See previous comment for resolution

File: pkg/config/config.go [hunk 7]
```diff
@@ -0,0 +1,660 @@
...
+
+	sourceType := lci.GetSourceType()
+	sourceLocation := lci.GetSourceLocation()
+
+	// Get the component context folder
+	// ".odo" is removed as lci.Filename will always return the '.odo' folder.. we don't need that!
+	componentContext := strings.Trim(filepath.Dir(lci.Filename), ".odo")
+
+	if sourceLocation == "" {
+		return "", fmt.Errorf("Blank source location provided")
```
[the remark this answers is not in the document]
Comment (rajivnathan) [reply pkg/config/config.go]: Looks like the message has actually changed in a newer copy of the file than the base commit for this fork so we can deal with that change when we decide which commit from odo should be the base for the kdo work.

**Pull request merged**